### DISCUSSION TITLE
안녕하세요 혹시 해당 프로젝트 오픈소스 기여 양식이 따로 없을까요?

### DISCUSSION THREAD

**Discussion by RudyBlack** [naver/egjs-view3d#45] [Q&A] [answered]
해당 오픈소스를 쓰고 있는데 post-processing 같은 기능을 따로 추가해서 테스트해보고 잘되면 pr 해서 기여 해보고 싶습니다. 따로 [CONTRIBUTING.md](https://github.com/naver/egjs-flicking/blob/master/CONTRIBUTING.md) 같은 양식이 있을까요??

**Comment by WoodNeck** [comment, marked as answer]
안녕하세요 :)

View3D는 현재 거의 저 혼자 작업하다보니 아직 컨트리뷰팅 가이드라인이 따로 마련되지 않은 상태입니다.
다만, egjs 컴포넌트들은 대부분의 환경설정들을 공유하고 있기 때문에
말씀해주신 CONTRIBUTING 가이드도 이 레포지토리에 따로 동일하게 적용하실 수 있습니다.

기본적으로는 다음 항목들만 지켜주시면 되구요, 코드는 PR 올려주시면 저희가 리뷰해드리니 부담없이 올려주시면 됩니다.
- Eslint & Editorconfig 사용 (예로, VS Code 사용시 아래 extension 설치 후 올바르게 동작하는지 확인하시면 됩니다)
![image](https://user-images.githubusercontent.com/26213435/202846967-5a0f1eae-e56c-4830-bcda-f6485bd785b8.png)

- 커밋 가이드라인 동일하게 사용
  - https://github.com/naver/egjs-flicking/blob/master/CONTRIBUTING.md
  - 항목 중 module은 보통 생략하고 있습니다.

기타 작업하시다가 궁금하신 점 생기시면 이 곳에 남겨주시는 등 어떤 형태로든 전달해주시면 답변 드리겠습니다.

**Reply by RudyBlack** [reply]
와우 빠르고 친절한 답변 감사합니다 :)

궁금한게 두가지 있는데요
첫째로 PR은 영어 한글 어떤게 더 편하실까요?
둘째로 잠시 훑어본 결과 packages/view3d 폴더 기준으로 작업 하는 것 같은데 수정사항을 라이브로 테스트 하려면 어떻게 해야 할까요?
<img width="553" alt="Screen Shot 2022-11-19 at 8 05 28 PM" src="https://user-images.githubusercontent.com/21374353/202847630-b4292f1c-0b83-4dad-b94b-d41803bc598f.png">

npm run dev 하면 
<img width="530" alt="Screen Shot 2022-11-19 at 8 06 08 PM" src="https://user-images.githubusercontent.com/21374353/202847644-7189429b-c894-4723-8bd1-256615b855fc.png">

이러한 결과가 나옵니다 ㅠㅠ

**Reply by WoodNeck** [reply]
저도 한국인이라 한글이 편하긴 한데 ㅎ
저희 컴포넌트들이 한국인만을 대상으로 하는게 아니라서 기록을 겸해서 가능하면 영어를 권해드리고 싶습니다.
어려우실 경우 파파고 등 번역기 사용하셔도 됩니다, 뜻만 통하면 되니까요

해당 dev 스크립트는 제가 초기에 잠깐 사용하고 지우는걸 깜빡한 것 같네요,
지금은 사용하지 않습니다.

지금 개발하시려면 데모 페이지를 직접 이용하시는게 편하실거에요
1. npm install을 다음 3 폴더에서 수행
- 프로젝트 루트, packages/view3d, demo
2. 프로젝트 루트에서 다음 스크립트 실행
```console
npm run docs:build
```
3. 데모 폴더에서 npm run start

demo에 있는 src/components/View3D.tsx에서 packages/view3d의 src를 참조하고 있기 때문에
packages/view3d 수정하시면서 작업해주시면 됩니다.

옵션 추가시에는 demo/docs/options 내에 다른 파일 참고하시어 mdx 파일을 추가해주시면 됩니다.

**Reply by RudyBlack** [reply]
Thank you. I'll try it now. If I don't know anything that, I'll ask you again!! :)
